Backend/BikesRentalServer.Services/MalfunctionsService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using BikesRentalServer.Infrastructure;
+using BikesRentalServer.Models;
+using BikesRentalServer.Services.Abstract;
+using BikesRentalServer.DataAccess.Repositories.Abstract;
+
+namespace BikesRentalServer.Services
+{
+    public class MalfunctionsService : IMalfunctionsService
+    {
+        private readonly UserContext _userContext;
+        private readonly IMalfunctionsRepository _malfunctionRepository;
+        private readonly IBikesRepository _bikesRepository;
+        private readonly IUsersRepository _usersRepository;
+        public MalfunctionsService(UserContext userContext, IMalfunctionsRepository malfunctionRepository, IBikesRepository bikesRepository, IUsersRepository usersRepository)
+        {
+            _malfunctionRepository = malfunctionRepository;
+            _userContext = userContext;
+            _bikesRepository = bikesRepository;
+            _usersRepository = usersRepository;
+        }
+
+        #region basics
+        public ServiceActionResult<Malfunction> AddMalfunction(string bikeId, string description)
+        {
+            var bike = _bikesRepository.Get(bikeId);
+            if (bike == null)
+                return ServiceActionResult.EntityNotFound<Malfunction>("Bike not found");
+            var user = _usersRepository.GetByUsername(_userContext.Username);
+
+            if(bike.User == null || (bike.User.Username != _userContext.Username))
```

Comment:
spacja po ifie, proponuję użyć `is null`, nawiasy nie są tu potrzebne (niemożliwym jest zrobienie logical or ze stringiem `bike.User.Username`)